core/src/test/java/nl/nn/adapterframework/util/MiscTest.java

```diff
@@ -1,43 +1,610 @@
+    public void testToSortName() throws Exception {
+assertEquals(Misc.toSortName("new_name"), "NEW*NAME");    }
+
+    /**
+     * Method: countRegex(String string, String regex)
+     */
+    @Test
+    public void testCountRegex() throws Exception {
+String s = "12ab34";
+    String regex = "\\d";
+    int regexCount = Misc.countRegex(s, regex);
+    assertEquals(regexCount, 4);
+    }
+
+    /**
+     * Method: resourceToString(URL resource)
+     */
+    @Test
+    public void testResourceToStringResource() throws Exception {
+        URL resource = new URL("http://example.com/");
+        String s1 = Misc.resourceToString(resource);
+        assertEquals(s1, "<!doctype html>\n" +
+                "<html>\n" +
+                "<head>\n" +
+                "    <title>Example Domain</title>\n" +
+                "\n" +
+                "    <meta charset=\"utf-8\" />\n" +
+                "    <meta http-equiv=\"Content-type\" content=\"text/html; charset=utf-8\" />\n" +
+                "    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\" />\n" +
+                "    <style type=\"text/css\">\n" +
+                "    body {\n" +
+                "        background-color: #f0f0f2;\n" +
+                "        margin: 0;\n" +
+                "        padding: 0;\n" +
+                "        font-family: -apple-system, system-ui, BlinkMacSystemFont, \"Segoe UI\", \"Open Sans\", \"Helvetica Neue\", Helvetica, Arial, sans-serif;\n" +
+                "        \n" +
+                "    }\n" +
+                "    div {\n" +
+                "        width: 600px;\n" +
+                "        margin: 5em auto;\n" +
+                "        padding: 2em;\n" +
+                "        background-color: #fdfdff;\n" +
+                "        border-radius: 0.5em;\n" +
+                "        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);\n" +
+                "    }\n" +
+                "    a:link, a:visited {\n" +
+                "        color: #38488f;\n" +
+                "        text-decoration: none;\n" +
+                "    }\n" +
+                "    @media (max-width: 700px) {\n" +
+                "        div {\n" +
+                "            margin: 0 auto;\n" +
+                "            width: auto;\n" +
+                "        }\n" +
+                "    }\n" +
+                "    </style>    \n" +
+                "</head>\n" +
+                "\n" +
+                "<body>\n" +
+                "<div>\n" +
+                "    <h1>Example Domain</h1>\n" +
+                "    <p>This domain is for use in illustrative examples in documents. You may use this\n" +
+                "    domain in literature without prior coordination or asking for permission.</p>\n" +
+                "    <p><a href=\"https://www.iana.org/domains/example\">More information...</a></p>\n" +
+                "</div>\n" +
+                "</body>\n" +
+                "</html>\n");
```

Comment:
In stead of using a big concatinated string, it might be easier and more readable to use `TestFileUtils.getResource` and read the contents from a file instead

Reply:
I assumed you meant `TestFileUtils.getTestFile` which uses `getResource`. So I'm using this method.